backend/controllers/userControllers.js

```diff
@@ -6,47 +6,61 @@ const register = async (request, response) => {
   if (username && password && email) {
     const salt = "batteryAnt"; //TODO find out what this is for...
     const imageUrl = "/"; //TODO: change this to the default image url
-    //TODO: check if this username is availabe in the db
-    let selectUserQuery =
-      "SELECT * FROM users WHERE username = $1 OR email = $2";
+    const selectUserQuery =
+      "SELECT * FROM users WHERE username = $1 or email = $2";
     try {
       const result = await connection.query(selectUserQuery, [username, email]);
+      let errors = [];
       if (result.length > 0) {
-        console.log(
-          `username: ${username} is already taken or email ${email} is already in use`,
-        );
-        response.send({ error: "username or email is already taken" });
-        return;
+        result.forEach((user) => {
+          if (user.username === username) {
+            errors.push("username is taken");
+          }
+
+          if (user.email === email) {
+            errors.push("email is taken");
+          }
+        });
+
+        return response.render("register", {
+          errors: errors.join("\n"),
+        });
```

Comment:
Resolves #25 

The client is now present with messages that specifies if its the username, email or both that are taken